primitives/transaction/src/account/vesting_contract.rs

```diff
@@ -68,79 +68,132 @@ impl AccountTransactionVerification for VestingContractVerifier {
     }
 }
 
-#[derive(Default, Clone, Debug, Serialize, Deserialize)]
+#[derive(Clone, Debug, Default)]
 pub struct CreationTransactionData {
+    pub owner: Address,
+    pub start_time: u64,
+    pub time_step: u64,
+    pub step_amount: Coin,
+    pub total_amount: Coin,
+}
+
+#[derive(Deserialize, Serialize, SerializedSize)]
+struct CreationTransactionData8 {
     pub owner: Address,
     #[serde(with = "nimiq_serde::fixint::be")]
+    #[serialize_size(fixed_size)]
+    pub time_step: u64,
+}
+#[derive(Deserialize, Serialize, SerializedSize)]
+struct CreationTransactionData24 {
+    pub owner: Address,
+    #[serde(with = "nimiq_serde::fixint::be")]
+    #[serialize_size(fixed_size)]
     pub start_time: u64,
     #[serde(with = "nimiq_serde::fixint::be")]
+    #[serialize_size(fixed_size)]
+    pub time_step: u64,
+    pub step_amount: Coin,
+}
+#[derive(Deserialize, Serialize, SerializedSize)]
+struct CreationTransactionData32 {
+    pub owner: Address,
+    #[serde(with = "nimiq_serde::fixint::be")]
+    #[serialize_size(fixed_size)]
+    pub start_time: u64,
+    #[serde(with = "nimiq_serde::fixint::be")]
+    #[serialize_size(fixed_size)]
     pub time_step: u64,
     pub step_amount: Coin,
     pub total_amount: Coin,
 }
 
 impl CreationTransactionData {
-    pub fn parse(transaction: &Transaction) -> Result<Self, TransactionError> {
-        let reader = &mut &transaction.recipient_data[..];
-        let (owner, left_over) = Deserialize::deserialize_take(reader)?;
-
-        if transaction.recipient_data.len() == Address::SIZE + 8 {
+    fn parse_impl(data: &[u8], tx_value: Coin) -> Result<Self, TransactionError> {
+        // TODO: Change this into `const` block once Rust 1.79 is sufficiently old.
+        #[allow(clippy::assertions_on_constants)]
+        {
+            assert!(CreationTransactionData8::SIZE != CreationTransactionData24::SIZE);
+            assert!(CreationTransactionData8::SIZE != CreationTransactionData32::SIZE);
+            assert!(CreationTransactionData24::SIZE != CreationTransactionData32::SIZE);
+        }
+        Ok(if data.len() == CreationTransactionData8::SIZE {
```

Comment:
Why are we not using here a `match` statement?

Reply:
Fixed.